Code/Framework/AzToolsFramework/Tests/ArchiveTests.cpp

```diff
@@ -132,69 +152,131 @@ namespace UnitTest
 #endif // AZ_TRAIT_DISABLE_FAILED_ARCHIVE_TESTS
         {
             QStringList fileList = CreateArchiveFileList();
 
             CreateArchiveFolder(GetArchiveFolderName(), fileList);
-
+            AZ_TEST_START_TRACE_SUPPRESSION;
             bool createResult = CreateArchive();
+            AZ_TEST_STOP_TRACE_SUPPRESSION_NO_COUNT;
 
             EXPECT_EQ(createResult, true);
 
             bool catalogCreated{ true };
             AZ::Test::AssertAbsorber assertAbsorber;
-            AzToolsFramework::AssetBundleCommandsBus::BroadcastResult(catalogCreated, &AzToolsFramework::AssetBundleCommandsBus::Events::CreateDeltaCatalog, GetArchivePath().toStdString().c_str(), true);
+            AzToolsFramework::AssetBundleCommandsBus::BroadcastResult(catalogCreated,
+                &AzToolsFramework::AssetBundleCommandsBus::Events::CreateDeltaCatalog, GetArchivePath().toStdString().c_str(), true);
 
             EXPECT_EQ(catalogCreated, false);
         }
 
 #if AZ_TRAIT_DISABLE_FAILED_ARCHIVE_TESTS
-        TEST_F(ArchiveTest, DISABLED_CreateDeltaCatalog_ArchiveWithoutCatalogAssetsRegistered_Success)
+        TEST_F(ArchiveComponentTest, DISABLED_AddFilesToArchive_FromListFile_Success)
 #else
-        TEST_F(ArchiveTest, CreateDeltaCatalog_ArchiveWithoutCatalogAssetsRegistered_Success)
+        TEST_F(ArchiveComponentTest, AddFilesToArchive_FromListFile_Success)
+#endif // AZ_TRAIT_DISABLE_FAILED_ARCHIVE_TESTS
+        {
+            QString listFile = CreateArchiveListTextFile();
+            CreateArchiveFolder(GetArchiveFolderName(), CreateArchiveFileList());
+
+            AZ_TEST_START_TRACE_SUPPRESSION;
+            std::future<bool> addResult;
+            AzToolsFramework::ArchiveCommandsBus::BroadcastResult(
+                addResult, &AzToolsFramework::ArchiveCommandsBus::Events::AddFilesToArchive, GetArchivePath().toStdString().c_str(),
+                GetArchiveFolder().toStdString().c_str(), listFile.toStdString().c_str());
+            bool result = addResult.get();
+            AZ_TEST_STOP_TRACE_SUPPRESSION_NO_COUNT;
+
+            EXPECT_TRUE(result);
+        }
+
+#if AZ_TRAIT_DISABLE_FAILED_ARCHIVE_TESTS
+        TEST_F(ArchiveComponentTest, DISABLED_ExtractArchive_AllFiles_Success)
+#else
+        TEST_F(ArchiveComponentTest, ExtractArchive_AllFiles_Success)
+#endif // AZ_TRAIT_DISABLE_FAILED_ARCHIVE_TESTS
+        {
+            CreateArchiveFolder();
+            AZ_TEST_START_TRACE_SUPPRESSION;
+            bool createResult = CreateArchive();
+            AZ_TEST_STOP_TRACE_SUPPRESSION_NO_COUNT;
+            EXPECT_TRUE(createResult);
+
+            AZ_TEST_START_TRACE_SUPPRESSION;
+            std::future<bool> extractResult;
+            AzToolsFramework::ArchiveCommandsBus::BroadcastResult(
+                extractResult, &AzToolsFramework::ArchiveCommandsBus::Events::ExtractArchive, GetArchivePath().toStdString().c_str(),
+                GetExtractFolder().toStdString().c_str());
```

Comment:
repeat: `toUtf8().constData()` should be used here

```suggestion
                extractResult, &AzToolsFramework::ArchiveCommandsBus::Events::ExtractArchive, GetArchivePath().toUtf8().constData(),
                GetExtractFolder().toUtf8().constData());
```